demos/composition/cyclow/es6/src/confirm.js

```diff
@@ -0,0 +1,34 @@
+import {Block} from 'cyclow'
+
+const viewConfirm = () => [
+  {tag: 'button', content: 'Confirm', on: {click: 'confirm'}},
+  {tag: 'button', content: 'Cancel', on: {click: 'cancel'}}
+]
+
+const viewDefault = mode => ({
+  tag: 'button',
+  content: 'Submit',
+  attrs: {disabled: mode === 'disabled'},
+  on: {click: 'maybe'}
+})
+
+const Confirm = () => Block({
+  inputs: ['disabled'],
+  outputs: ['confirmation'],
+  on: {
+    init: () => state => ({mode: 'disabled'}),
+    disabled: (disabled, {mode}) => {
+      if (disabled) {
+        return state => ({mode: 'disabled'})
+      } else if (mode === 'disabled') {
+        return state => ({mode: 'waiting'})
+      }
+    },
+    maybe: () => state => ({mode: 'confirm'}),
+    cancel: () => state => ({mode: 'waiting'}),
+    confirm: () => [state => ({mode: 'waiting'}), ['out.confirmation']]
```

Comment:
What if I need access to the current state to calculate an event message or data? Is there an optional syntax where instead of `[eventname,payload]`, I can pass in a function that returns the event message array, like `state => [eventname,payload]`? 

Or maybe just for the data, so `[eventname, state => payload]` instead of `[eventname, payload]`?